src/queries/role.ts

```diff
@@ -63,8 +64,9 @@ const queryAllByRole = (
     return findAll(
       instance,
       (node) =>
-        // run the cheapest checks first, and early exit too avoid unneeded computations
+        // run the cheapest checks first, and early exit to avoid unneeded computations
         typeof node.type === 'string' &&
+        isAccessibilityElement(node) &&
```

Comment:
This would be a breaking change so we should apply it only when `useBreakingChanges` config is set.